src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1989,6 +1989,7 @@ bool os::pd_release_memory(char* addr, size_t size) {
   // Dynamically do different things for mmap/shmat.
   vmembk_t* const vmi = vmembk_find(addr);
   guarantee0(vmi);
+  vmi->assert_is_valid_subrange(addr, size);
```

Comment:
Reviewer note: this is unrelated to the problem, I just lifted the assert call from both branches and made it independent from debug/release.